File: 프로그래머스/1/42862. 체육복/체육복.cpp
// 그리디
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = n - lost.size();
    
    for(int i=0; i<reserve.size(); i++){
        for(int j=0; j<lost.size(); j++){
            if(reserve[i] == lost[j]){
                lost.erase(lost.begin()+j);
                reserve.erase(reserve.begin()+i);
                i--, j--;
                answer++;
            }
        }
    }
    
    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    
    for(int i=0; i<reserve.size(); i++){
        for(int j=0; j<lost.size(); j++){
            if(reserve[i]-1 == lost[j] || reserve[i]+1 == lost[j]){
                lost.erase(lost.begin()+j);
                reserve.erase(reserve.begin()+i);
                answer++;
                i--, j--;
                break;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    
    let lost_student = lost.filter((student) => !reserve.includes(student)).sort((a, b) => a-b);
    let reserve_student = reserve.filter((student) => !lost.includes(student)).sort((a, b) => a-b);
    
    answer += n-lost_student.length;
    
    for(let i=0; i<reserve_student.length; i++){
        for(let j=0; j<lost_student.length; j++){
            if(reserve_student[i]-1 === lost_student[j] || reserve_student[i]+1 === lost_student[j]){
                lost_student[j] = -1;
                reserve_student[i] = -1;
                answer++;
                break;
            }
        }
    }
    
    return answer;
}